File: backend/semgrep_rules/tests/python/kisa_sw_32_download_without_integrity.py
import requests
def unsafe(url):
    # ruleid: kisa.sw32.python.downloaded-code-executed-review
    exec(requests.get(url).text)
def safe(url):
    code = requests.get(url).content
    # hash/signature verification omitted in this fixture
    # ok: kisa.sw32.python.downloaded-code-executed-review
    return code

File: backend/semgrep_rules/tests/javascript/kisa_sw_32_download_without_integrity.js
const axios = require("axios");
async function unsafe(url) {
  // ruleid: kisa.sw32.javascript.downloaded-code-executed-review
  return eval((await axios.get(url)).data);
}
async function safe(url) {
  const data = (await axios.get(url)).data;
  // ok: kisa.sw32.javascript.downloaded-code-executed-review
  return data;
}
